packages/calcite-components/src/tests/commonTests.ts

```diff
@@ -1839,3 +1841,262 @@ export function openClose(componentTagOrHTML: TagOrHTML, options?: OpenCloseOpti
+            const shadowSelectors = shadowSelector.split(" ");
+
+            for (let i = 0; i < shadowSelectors.length; i++) {
+              const s = shadowSelectors[i];
+
+              if (i === 0) {
+                target = await page.find(`${selector} >>> ${s}`);
+              } else if (target && shadowSelectors[i + 1] === ">>>") {
+                target = await target.find(`${s} >>> ${shadowSelectors[i + 2]}`);
+                i += 2;
+              } else if (target) {
+                target = await target.find(s);
+              }
+            }
+          } else {
+            target = shadowSelector ? await page.find(`${selector} >>> ${shadowSelector}`) : target;
+          }
+        }
+        if (state && typeof state !== "string") {
+          contextSelector = Object.values(state)[0];
+        }
+
+        testTargets.push({ target, targetProp, contextSelector, state: stateName, expectedValue: setTokens[token] });
+      }
+    }
+
+    // set style attribute on styleTargets with the assigned token values
+    for (const selector in styleTargets) {
+      const [el, assignedCSSVars] = styleTargets[selector];
+
+      // Sets the style of each element to a string of CSS token props with themed token values
+      el.setAttribute("style", assignedCSSVars.join("; "));
+    }
+
+    await page.waitForChanges();
+
+    // Assert target computedStyle targetProp matches test theme token color
+    for (let i = 0; i < testTargets.length; i++) {
+      await assertThemedProps(page, { ...testTargets[i] });
+    }
+  });
+}
+
+export type ContextSelectByAttr = { attribute: string; value: string | RegExp };
+
+/**
+ * Custom type describing a test target for themed components. Use with themed and assertThemedProps.
+ */
+export type TestTarget = {
+  target: E2EElement;
+  contextSelector?: string | ContextSelectByAttr;
+  targetProp: keyof CSSStyleDeclaration;
+  state?: string;
+  expectedValue: string;
+};
+
+/**
+ * Custom type describing a test selector for themed components. Use with themed assertThemedProps.
+ */
+export type TestSelectToken = {
+  selector: string;
+  shadowSelector?: string;
+  targetProp: keyof CSSStyleDeclaration;
+  state?: string | Record<string, ContextSelectByAttr>;
+};
+
+/**
+ * Get the computed style of an element and assert that it matches the expected themed token value.
+ * This is useful for testing themed components.
+ *
+ * @param page - the e2e page
+ * @param options - the options to pass to the utility
+ * @param options.target - the element to get the computed style from
+ * @param options.contextSelector - the selector of the target element
+ * @param options.targetProp - the CSSStyleDeclaration property to check
+ * @param options.state - the state to apply to the target element
+ * @param options.expectedValue - the expected value of the targetProp
+ */
+export async function assertThemedProps(page: E2EPage, options: TestTarget): Promise<void> {
```

Comment:
Does this need to be exported?

Reply:
why wouldn't we export it? It might be useful in other contexts and costs us nothing to export.